File: app/api/schemas/jobs.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class JobResponse(BaseModel):
    id: int
    document_version_id: int
    document_name: str
    job_type: str
    status: str
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class JobListResponse(BaseModel):
    jobs: list[JobResponse]
    total: int
    pending_count: int
    running_count: int
    completed_count: int
    failed_count: int
    cancelled_count: int


class JobActionResponse(BaseModel):
    success: bool
    message: str
    job: JobResponse


class QueueActionResponse(BaseModel):
    success: bool
    message: str
    messages_cleared: int


# Hierarchical job schemas
class ChildJobCounts(BaseModel):
    total: int
    completed: int
    failed: int


class ParentJobResponse(BaseModel):
    id: int
    document_version_id: int
    document_name: str
    document_version: str
    job_type: str
    status: str
    error_message: Optional[str] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    child_job_counts: ChildJobCounts

    class Config:
        from_attributes = True


class ParentJobListResponse(BaseModel):
    jobs: list[ParentJobResponse]
    total: int
    pending_count: int
    running_count: int
    completed_count: int
    failed_count: int
    cancelled_count: int


class ChildJobResponse(BaseModel):
    id: int
    job_type: str
    status: str
    error_message: Optional[str] = None
    chunk_start_index: Optional[int] = None
    chunk_end_index: Optional[int] = None
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class JobDetailResponse(BaseModel):
    parent_job: ParentJobResponse
    child_jobs: list[ChildJobResponse]

File: app/ui/src/types/jobs.ts
export type JobStatus =
  | "pending"
  | "running"
  | "completed"
  | "failed"
  | "cancelled";

export interface Job {
  id: number;
  document_version_id: number;
  document_name: string;
  job_type: string;
  status: JobStatus;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface JobListResponse {
  jobs: Job[];
  total: number;
  pending_count: number;
  running_count: number;
  completed_count: number;
  failed_count: number;
  cancelled_count: number;
}

export interface JobActionResponse {
  success: boolean;
  message: string;
  job: Job;
}

export interface QueueActionResponse {
  success: boolean;
  message: string;
  messages_cleared: number;
}

// Hierarchical job types
export interface ChildJobCounts {
  total: number;
  completed: number;
  failed: number;
}

export interface ParentJob {
  id: number;
  document_version_id: number;
  document_name: string;
  document_version: string;
  job_type: string;
  status: JobStatus;
  error_message: string | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
  child_job_counts: ChildJobCounts;
}

export interface ChildJob {
  id: number;
  job_type: string;
  status: JobStatus;
  error_message: string | null;
  chunk_start_index: number | null;
  chunk_end_index: number | null;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ParentJobListResponse {
  jobs: ParentJob[];
  total: number;
  pending_count: number;
  running_count: number;
  completed_count: number;
  failed_count: number;
  cancelled_count: number;
}

export interface JobDetailResponse {
  parent_job: ParentJob;
  child_jobs: ChildJob[];
}
